clients/unshaded/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -61,7 +61,10 @@ public static BlockOutStream get(TachyonFile tachyonFile, WriteType opType, int
 
     if (tachyonFile.mTachyonFS.hasLocalWorker()
         && tachyonConf.getBoolean(Constants.USER_ENABLE_LOCAL_WRITE,
-            Constants.DEFAULT_USER_ENABLE_LOCAL_WRITE)) {
+            Constants.DEFAULT_USER_ENABLE_LOCAL_WRITE)
+            && tachyonConf.get(
```

Comment:
Yes I agree with you on this point. Case the local worker is chosen we need to use LocalBlockOutStream. I still need some changes to take it into account.
The change I have added is to avoid a failure when we choose a RemoteWorker and there is a Local one because the previous condition didn't take it into account.
I hope I make my explanation clearer